### PR TITLE
AQM.v7 EE2 compliance: -ftrapuv and -check all in DEBUG build options

### PR DESCRIPTION
**Pull request by BrianCurtis-NOAA**
This PR replaces all `-check` build flags with `-check all` for EE2 compliance for AQM v7

### CODE REVIEW

**Comment by BrianCurtis-NOAA** [comment]
@aerorahul i don't know who officially would look this over and give an OK as I believe this fork is only for production needs. Would you be able to give an OK? It looks like I will be able to hit the merge button myself.

File: cmake/compiler_flags_Intel_Fortran.cmake [hunk 1]
```diff
@@ -10,7 +10,7 @@ set(CMAKE_Fortran_FLAGS_REPRO "-O2 -debug minimal -fp-model consistent -qoverrid
 
 set(CMAKE_Fortran_FLAGS_RELEASE "-O2 -debug minimal -fp-model source -qoverride-limits -qopt-prefetch=3")
 
-set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -check -check noarg_temp_created -check nopointer -warn -warn noerrors -fp-stack-check -fstack-protector-all -fpe0 -debug -ftrapuv")
+set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -check all -check noarg_temp_created -warn -warn noerrors -fp-stack-check -fstack-protector-all -fpe0 -debug -ftrapuv")
```
Comment (aerorahul) [inline cmake/compiler_flags_Intel_Fortran.cmake:13]: I think you can remove `-check noarg_temp_created` when `-check all` is added.
c.f. [check](https://www.intel.com/content/www/us/en/docs/fortran-compiler/developer-guide-reference/2024-0/check.html)

Comment (BrianCurtis-NOAA) [reply cmake/compiler_flags_Intel_Fortran.cmake:13]: We want all options enabled with `-check all` but err files were on the order of many GB, so we disabled the check for arg_temp_created by adding `-check noarg_temp_created`.

Comment (aerorahul) [reply cmake/compiler_flags_Intel_Fortran.cmake:13]: Its not like the `debug` build is kept around for a long time.  But Ok.  Thanks for clarifying.

**Review by aerorahul — APPROVED** [review]
oh yeah. sorry. LGTM.

**Pull request merged**